crates/bvs-vault-router/src/contract.rs

```diff
@@ -326,13 +331,156 @@ mod execute {
+            false => {
+                return Err(ContractError::InvalidSlashingRequest {
+                    msg: "Collateral already locked for this id".to_string(),
+                })
+            }
+        }
+
+        if slash_req.status != SlashingRequestStatus::Pending as u8 {
+            return Err(ContractError::InvalidSlashingRequest {
+                msg: "Slashing request is not pending".to_string(),
+            });
+        }
+
+        let SlashingParametersResponse(slashing_parameters) = deps.querier.query_wasm_smart(
+            registry.clone(),
+            &bvs_registry::msg::QueryMsg::SlashingParameters {
+                service: info.sender.to_string(),
+                timestamp: Some(slash_req.request.timestamp.seconds()),
+            },
+        )?;
+
+        let slashing_parameters = match slashing_parameters {
+            Some(p) => p,
+            None => {
+                return Err(InvalidSlashingRequest {
+                    msg: "Service has not enabled slashing at timestamp.".to_string(),
+                })
+            }
+        };
+
+        let accused_operator = deps
+            .api
+            .addr_validate(slash_req.request.operator.as_str())?;
+
+        let true_id = state::SLASHING_REQUEST_IDS
+            .may_load(deps.storage, (&info.sender, &accused_operator))?;
+
+        // Check if the id is the same as the one in the request
+        if true_id != Some(id.clone()) {
+            return Err(ContractError::Unauthorized {
+                msg: "Slash locking is restricted to the service that initiated the request."
+                    .to_string(),
+            });
+        }
+
+        let now = env.block.time;
+
+        if now > slash_req.request_expiry {
+            return Err(ContractError::InvalidSlashingRequest {
+                msg: "Slashing has expired".to_string(),
+            });
+        };
+
+        if now
+            < slash_req
+                .request_time
+                .plus_seconds(slashing_parameters.resolution_window)
+        {
+            return Err(ContractError::InvalidSlashingRequest {
+                msg: "Resolution window for this slashing has not passed".to_string(),
+            });
+        };
+
+        let vaults_managed = state::OPERATOR_VAULTS
+            .prefix(&accused_operator)
+            .keys(deps.storage, None, None, cosmwasm_std::Order::Ascending)
+            .collect::<StdResult<Vec<_>>>()?;
+
+        // Array instead of map to avoid the overheads
+        let mut messages = Vec::new();
+        for vault in vaults_managed {
+            let vault_info = vault::get_vault_info(deps.as_ref(), &vault)?;
+            let total_assets = vault_info.total_assets;
+            let bips = Uint128::from(slash_req.request.bips as u128);
+
+            // Due to the nature of the integer division involved, the result is always floored.
+            let slash_absolute = total_assets.multiply_ratio(bips, Uint128::from(10000_u128));
```

Comment:
i think type conversion from `u16` to `u128` is not necessary as `.multiply_ratio` will convert it into `u128` in the underlying code. 

Also no need to convert u128 to Uint128 when it will be converted back to u128
```suggestion
            // Due to the nature of the integer division involved, the result is always floored.
            let slash_absolute = total_assets.multiply_ratio(slash_req.request.bips, 10000_u128);
```